### ISSUE TITLE
Document getting started?

### ISSUE THREAD

**Issue by msabramo** [iodide-project/pyodide#338]
Apologies in advance if I overlooked the appropriate docs...

Pyodide looks very interesting from the description and from looking at the live demo.

I've been wanting to tinker around with it, but have been unable to get anything interesting to run locally. I cloned the repo, ran the Docker container, did `make` and after a long time, it seemed to succeed and output a `build` dir with lots of JS and HTML files. So far so good!

So then I launched a little mini web server to host the files (because I saw a comment somewhere warning that opening the HTML files directly wouldn't work because of CORS restrictions) and I tried to to view `test.html`, `python.html`, and `matplotlib.html` in my browser from that local server, but all I get are blank pages with errors. Those pages seem to reference CSS and JS files on the remote site and some of those don't work anymore and redirect. It seems like it a local build would ideally be self-contained and would have only links to locally built resources, but that doesn't seem to be happening.

Here's an example page that I get:

<img width="880" alt="Screen Shot 2019-03-14 at 6 21 17 PM" src="https://user-images.githubusercontent.com/305268/54401732-14339e80-4686-11e9-9fbb-73081a2e1db9.png">

With some amount of hacking the HTML files to point to what may or may not be the correct CSS and JS files locally, I've managed to get the errors to go away and to get still not terribly impressive output like this:
 
<img width="496" alt="Screen Shot 2019-03-14 at 6 23 20 PM" src="https://user-images.githubusercontent.com/305268/54401767-46dd9700-4686-11e9-8d36-5345a403244f.png">

Maybe it is initializing Python successfully and I need to do other stuff to use it, but I don't know what the next steps are.

I'm wondering if it might be possible to add a few lines to the README showing newbies how to get some interesting output - something that shows that things are working and perhaps inspires people to use it more and maybe even contribute to the project?

**Comment by msabramo** [comment]
By the way, I also discovered https://github.com/iodide-project/pyodide-demo from browsing some other issues and I tried to launch a little web server in a fresh checkout of that and then view that and I get very similar results to what I got from building myself:

```
$ git clone git@github.com:iodide-project/pyodide-demo.git
Cloning into 'pyodide-demo'...
remote: Enumerating objects: 62, done.
remote: Total 62 (delta 0), reused 0 (delta 0), pack-reused 62
Receiving objects: 100% (62/62), 39.38 MiB | 13.05 MiB/s, done.
Resolving deltas: 100% (12/12), done.

$ cd pyodide-demo

   ┌───────────────────────────────────────────────────┐
   │                                                   │
   │   Serving!                                        │
   │                                                   │
   │   - Local:            http://localhost:5000       │
   │   - On Your Network:  http://192.168.86.45:5000   │
   │                                                   │
   │   Copied local address to clipboard!              │
   │                                                   │
   └───────────────────────────────────────────────────┘
```

resulting in:

<img width="571" alt="Screen Shot 2019-03-14 at 6 30 05 PM" src="https://user-images.githubusercontent.com/305268/54401948-38dc4600-4687-11e9-888a-df7589b2ab35.png">

so maybe I'm doing something wrong or maybe some part of the process doesn't work the way that it used to?

**Comment by msabramo** [comment]
The `Uncaught SyntaxError: Unexpected token <` error is really weird, because it's on the first line of the HTML which is the `<!DOCTYPE html>` declaration and I don't understand why that would give a syntax error.

But I did see some comments on the Interwebz about that possibly being somehow related to having CSS or JS that's not loading, so I started messing around with the links.

```diff
diff --git a/python.html b/python.html
index cf58601..ea8072b 100644
--- a/python.html
+++ b/python.html
@@ -3,7 +3,7 @@
 <head>
 <meta charset="UTF-8">
 <title>Python - iodide</title>
-<link rel="stylesheet" type="text/css" href="https://iodide.io/dist/iodide.pyodide-20180623.css">
+<link rel="stylesheet" type="text/css" href="renderedhtml.css">
 </head>
 <body>
 <script id="jsmd" type="text/jsmd">
@@ -262,6 +262,6 @@ A couple things that already work that will be coming to this example notebook s
 %% js
 </script>
 <div id='page'></div>
-<script src='https://iodide.io/dist/iodide.pyodide-20180623.js'></script>
+<script src='pyodide.js'></script>
 </body>
 </html>
```

and that gets me this output:

<img width="950" alt="Screen Shot 2019-03-14 at 6 36 08 PM" src="https://user-images.githubusercontent.com/305268/54402128-20b8f680-4688-11e9-8105-9d63ca39b639.png">

If I instead use `pyodide_dev.js`:

```diff
diff --git a/python.html b/python.html
index cf58601..b2b5692 100644
--- a/python.html
+++ b/python.html
@@ -3,7 +3,7 @@
 <head>
 <meta charset="UTF-8">
 <title>Python - iodide</title>
-<link rel="stylesheet" type="text/css" href="https://iodide.io/dist/iodide.pyodide-20180623.css">
+<link rel="stylesheet" type="text/css" href="renderedhtml.css">
 </head>
 <body>
 <script id="jsmd" type="text/jsmd">
@@ -262,6 +262,6 @@ A couple things that already work that will be coming to this example notebook s
 %% js
 </script>
 <div id='page'></div>
-<script src='https://iodide.io/dist/iodide.pyodide-20180623.js'></script>
+<script src='pyodide_dev.js'></script>
 </body>
 </html>
```

then I get the somewhat better but still not exciting:

<img width="950" alt="Screen Shot 2019-03-14 at 6 36 08 PM" src="https://user-images.githubusercontent.com/305268/54402210-6b3a7300-4688-11e9-865d-df7a2100a699.png">

Chrome Dev Tools shows that it's loading a bunch of interesting-looking resources:

<img width="1148" alt="Screen Shot 2019-03-14 at 6 39 51 PM" src="https://user-images.githubusercontent.com/305268/54402267-9e7d0200-4688-11e9-91a5-d3217676baaa.png">

but somehow I feel like if it were truly working perfectly, then I should be seeing something more interesting...?

**Comment by msabramo** [comment]
Hmmm, looking at `pyodide.js` for inspiration, I messed around and got some interesting things to work:

<img width="572" alt="Screen Shot 2019-03-14 at 6 48 06 PM" src="https://user-images.githubusercontent.com/305268/54402606-d20c5c00-4689-11e9-8178-8439d5ac59b4.png">

So perhaps I was just expecting something fancier like the iodide demo, but maybe this is more of a base component that just lets you run Python code, but I think it would be helpful if the docs outlined what you can do with this. I think if the docs show a bit of what can be done, then people will be more likely to play with it more and stretch the boundaries and then maybe contribute to the project, which is good for everyone.

**Comment by msabramo** [comment]
Here's a simple example of using numpy:

<img width="447" alt="Screen Shot 2019-03-14 at 6 57 57 PM" src="https://user-images.githubusercontent.com/305268/54402910-3845ae80-468b-11e9-9ede-c1af9b4963b2.png">

If these are reasonable examples of how pyodide is intended to be used, then I am happy to submit a PR that adds them to the `README.md`, but maybe I'm not using it as intended so I'll wait for some confirmation.

**Comment by msabramo** [comment]
I tried to use `pandas` but so far that's failing.

<img width="1680" alt="Screen Shot 2019-03-14 at 8 31 50 PM" src="https://user-images.githubusercontent.com/305268/54406695-49e18300-4698-11e9-9d76-f919828af271.png">

I'm not sure how to use `WebAssembly.compile` yet and I did try running `python setup.py build_ext --inplace --force` in the `/src/packages/pandas/build/pandas-0.23.4` dir in the Docker container but that didn't seem to help.

**Comment by msabramo** [comment]
Hmmm I'm not sure what happened but it mysteriously started working:

<img width="1680" alt="Screen Shot 2019-03-14 at 8 50 07 PM" src="https://user-images.githubusercontent.com/305268/54407388-db51f480-469a-11e9-8128-27e1085a059c.png">

This is looking pretty interesting. I think adding some simple commands like this to the docs will help people get started with pyodide more easily.

**Comment by msabramo** [comment]
Oops. Somehow I managed to completely overlook https://github.com/iodide-project/pyodide/tree/master/docs and especially https://github.com/iodide-project/pyodide/blob/master/docs/using_pyodide_from_javascript.md

Sorry about that.

It might help to make some of that info more prominent on the README page. I probably overlooked it because it was a link in a bunch of text and my eyes were probably searching for something that looked like code that I could run. So I'd contend that having one or two examples in the README would help with discoverability and you can still link to the separate docs.

**Comment by Kreijstal** [comment]
You just documented it!

**Comment by mdboom** [comment]
The tl;dr here is that the old method that was used for building a lot of the demos (standalone Iodide notebooks) has gone away in the Iodide project.  We have plans for a replacement that would load a minimal local Pyodide environment into a remote-hosted Iodide notebook, but that's taking a lot longer than we expected.  In the meantime, the documenting the best path has been neglected, but it seems like you got to it.  Basically, there isn't a standalone demo right now to run locally right now, but it can be imported and poked at from the browser console.

Personally, I run my own local copy of Iodide, and put a locally built Pyodide in there.  Or, if I'm just trying to confirm that something works, I write a unit test and let selenium take care of running it for me.

**Comment by jeff-hykin** [comment]
I had the same issue @msabramo, I would have started trying to reverse engineer the linked demo had I not found this issue.

@Kreijstal I still don't see it in the README.md near getting started

**Comment by rth** [comment]
@jeff-hykin Thanks for the feedback! Just to double check you are saying that it's hard to find the link to the documentation https://pyodide.readthedocs.io/en/latest/ in the readme? A agree the readme can certainly be improved, and we should probably move "building from sources" to the doc. 

Or are you saying that an example of usage from JS would be helpful in the readme as well?

**Comment by jeff-hykin** [comment]
I'm saying its hard to find how to get started from the readme.

Have something like this in the readme

> ### Getting Started
> 1. Here's how to get started using Pyodide in Javascript:  https://github.com/iodide-project/pyodide/blob/master/docs/using_pyodide_from_javascript.md
> 2. You can also build Pyodide from source (link)
> 3. You can also use docker (link)

"Getting started" doesn't really count if by the end of it I still don't even have the most basic demo-able use of pyodide. The JS documentation is great and is effectively the true getting started guide in my opinion. The current one is just download/build instructions.
